File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common/Connection/DiscoveryDocumentProviderTests.fs
namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common

open AuthZI.MicrosoftEntra
open Xunit
open Xunit.Abstractions

type DiscoveryDocumentProviderTestsBase(output: ITestOutputHelper) =
  [<Fact>]
  member _.``The system can obtain Discovery Document from Microsoft Entra ID endpoint``() =
    async {
      // Arrange
      let discoveryDocumentProvider = DiscoveryDocumentProvider(TestData.Web1ClientApp.DiscoveryEndpointUrl)

      let! discoveryDocument = discoveryDocumentProvider.GetDiscoveryDocumentAsync() |> Async.AwaitTask

      if discoveryDocument.IsSome then
        output.WriteLine(discoveryDocument.Value.DiscoveryEndpoint)

      // Act
      Assert.True(discoveryDocument.IsSome)
    }

File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common.CSharp/Connection/DiscoveryDocumentProviderTests.cs
using System.Threading.Tasks;
using AuthZI.MicrosoftEntra;
using Microsoft.FSharp.Core;
using Xunit;

namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common;

public class DiscoveryDocumentProviderTestsBase(ITestOutputHelper output)
{
  [Fact]
  public async Task TheSystemCanObtainDiscoveryDocumentFromMicrosoftEntraIDEndpoint()
  {
    var discoveryDocumentProvider = new DiscoveryDocumentProvider(TestData.Web1ClientApp.DiscoveryEndpointUrl);

    var discoveryDocument = await discoveryDocumentProvider.GetDiscoveryDocumentAsync();
    var hasDiscoveryDocument = FSharpOption<DiscoveryDocument>.get_IsSome(discoveryDocument);

    if (hasDiscoveryDocument)
    {
      output.WriteLine(discoveryDocument.Value.DiscoveryEndpoint);
    }

    Assert.True(hasDiscoveryDocument);
  }
}
